pandas/core/arrays/_mixins.py

```diff
@@ -304,6 +304,68 @@ def __repr__(self) -> str:
     # ------------------------------------------------------------------------
     # __array_function__ methods
 
+    def __array_function__(self, func, types, args, kwargs):
```

Comment:
array_ufunc now exists in pandas/core/arraylike.py we should share as much as possible here (might mean breaking up the existing code).

Reply:
I think the only shared code will be the checking of the types. We've standardized on a pattern like

```
class Foo:
    _HANDLED_TYPES = (...,)
```

And then ensuring that the set of types in `types` is a subset of `_HANDLED_TYPES + (type(self),)`.